_docs/_latest/setup/upgrade.md

```diff
@@ -1208,6 +1208,100 @@ Example command: `gcloud compute instances reset forseti-server-vm-70ce82f --zon
+You can reset the VM by running command `gcloud compute instances reset MY_FORSETI_SERVER_INSTANCE --zone MY_FORSETI_SERVER_ZONE` 
+Example command: `gcloud compute instances reset forseti-server-vm-70ce82f --zone us-central1-c`
+1. Repeat step `3-9` for Forseti client.
+1. Configuration file `forseti_conf_server.yaml` updates: 
+   **Inventory**
+   - Update the `inventory` to include support for `composite_root_resources`.
+      ```
+        scanner:
+            ...
+            scanners:
+                - name: config_validator
+                  enabled: false
+                - name: groups_settings
+                  enabled: true
+            ...
+        ##################################################
+        notifier:
+            ...
+            resources:
+                - resource: config_validator_violations
+                  should_notify: true
+                  notifiers:
+                    # Email violations
+                    - name: email_violations
+                    # Upload violations to GCS.
+                    - name: gcs_violations
+                      configuration:
+                        data_format: csv
+                        # gcs_path should begin with "gs://"
+                        gcs_path: gs://{FORSETI_BUCKET}/scanner_violations
+                - resource: groups_settings_violations
+                  should_notify: true
+                  notifiers:
+                    # Email violations
+                    - name: email_violations
+                    # Upload violations to GCS.
+                    - name: gcs_violations
+                      configuration:
+                        data_format: csv
+                        # gcs_path should begin with "gs://"
+                        gcs_path: gs://{FORSETI_BUCKET}/scanner_violations
+                ...
+       ```
+
+1. Rule files updates:
+  - Update [KMS rule file](https://github.com/GoogleCloudPlatform/forseti-security/blob/v2.14.0/rules/kms_rules.yaml)
+    under `rules/` in your Forseti server GCS bucket to include sample rules according to CIS benchmark.
+  - Add [Groups settings rule file](https://github.com/GoogleCloudPlatform/forseti-security/blob/v2.14.0/rules/groups_settings_rules.yaml)
+    under `rules/` in your Forseti server GCS bucket to include Groups Settings rules.
+### Steps to upgrade using Terraform
+
+1. Update the `version` inside `main.tf` file to `1.4.0`.
+1. Run command `terraform plan` to see the infrastructure plan.
+1. Run command `terraform apply` to apply the infrastructure build.
+1. Rule files updates:
+  - Update [KMS rule file](https://github.com/GoogleCloudPlatform/forseti-security/blob/v2.14.0/rules/kms_rules.yaml)
+    under `rules/` in your Forseti server GCS bucket to include sample rules according to CIS benchmark.
+  - Add [Groups settings rule file](https://github.com/GoogleCloudPlatform/forseti-security/blob/v2.14.0/rules/groups_settings_rules.yaml)
```

Comment:
You don't need to do this if you run `terraform init` --> It should ick up new files. It will not edit pre-existing files.

Reply:
removed